src/js/components/conversation.jsx

```diff
@@ -54,34 +65,145 @@ export class ConversationComponent extends Component {
 
     };
 
+    onScroll = () => {
+        const {dispatch, shouldScrollToBottom} = this.props;
+
+        // If top of Conversation component is reached, we need to fetch older messages
+        const node = findDOMNode(this);
+        if(node.scrollTop === 0) {
+            this.fetchHistory();
+        } else if(shouldScrollToBottom) {
+            dispatch(setShouldScrollToBottom(false));
+        }
+    };
+
+    fetchHistory = () => {
+        const {dispatch, hasMoreMessages, isFetchingMoreMessages, messages} = this.props;
+
+        const node = findDOMNode(this);
+        if(hasMoreMessages && !isFetchingMoreMessages) {
+            // make sure the last message is one from the server, otherwise it doesn't need to scroll to previous first message
+            if (messages.length > 0 && messages[messages.length - 1]._id) {
+                this._lastTopMessageId = messages[0]._id;
+            }
+            
+            const top = getTop(this._topMessageNode, node);
+            this._lastTopMessageNodePosition = top - node.scrollTop;
+            dispatch(setFetchingMoreMessages(true));
+            setTimeout(() => {
```

Comment:
I think this was originally so that the transition from the `Load more` to `Retrieving history` text appears less choppy. But now, looks like it doesn't actually make a difference... I'll remove this